### PR TITLE
bech32: Add DecodeUnsafe for bypassing checksum validation

### PR DESCRIPTION
**Pull request by conr2d**
Fixes cosmos/cosmos-sdk#10025.

### CODE REVIEW

File: bech32/bech32.go [hunk 1]
```diff
@@ -229,17 +261,17 @@ func DecodeNoLimit(bech string) (string, []byte, error) {
...
 
 // Decode decodes a bech32 encoded string, returning the human-readable part and
 // the data part excluding the checksum.
 //
 // Note that the returned data is 5-bit (base32) encoded and the human-readable
 // part will be lowercase.
-func Decode(bech string) (string, []byte, error) {
-	// The maximum allowed length for a bech32 string is 90.
-	if len(bech) > 90 {
+func Decode(bech string, limit int) (string, []byte, error) {
```
Comment (ValarDragon) [inline bech32/bech32.go:272]: This change is the same as the enigma change which the SDK already uses IIRC, so not API breaking for the cosmos SDK

**Review by ValarDragon — COMMENTED** [review]
LGTM!

**Comment by tac0turtle** [comment]
Hey, thanks for opening this PR. IN the future lets get reviews prior to merging. Seems tests arent passing now

**Comment by conr2d** [comment]
@marbar3778 I'm sorry for merging PR before getting review. I thought that ValarDragon's comment meant LGTM, but he had forgotten approving PR. I checked whether all unit tests pass, but missed linting things. (gofmt) There is no functional issue.

@ValarDragon Thank you for reviewing this.

**Pull request merged**